trestle/core/commands/transform_oscal.py

```diff
@@ -0,0 +1,93 @@
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Trestle transform command for oscal files."""
+import argparse
+import logging
+import pathlib
+import traceback
+
+import trestle.core.const as const
+import trestle.utils.log as log
+from trestle.core.commands.author.ssp import SSPFilter
+from trestle.core.commands.command_docs import CommandPlusDocs
+from trestle.core.profile_resolver import ProfileResolver
+from trestle.oscal.profile import Profile
+from trestle.utils import fs
+
+logger = logging.getLogger(__name__)
+
+
+class TransformCmd(CommandPlusDocs):
+    """Transform an oscal model or models by filtering or aggregation."""
+
+    name = 'transform'
+
+    def _init_arguments(self) -> None:
+        self.add_argument('-t', '--type', help='Input model type', choices=const.MODEL_TYPE_LIST, required=True)
+        self.add_argument('-tf', '--transform', help='Type of transform', choices=const.TRANSFORM_TYPES, required=True)
+        self.add_argument(
+            '-tb',
+            '--transform-by',
+            type=str,
+            help='Model name or comma-separated item names used in the transform',
+            required=False,
+            default=''
+        )
+        self.add_argument('-i', '--input', help='Name of input model', type=str, required=True)
+        self.add_argument('-o', '--output', help='Name of output model or directory', type=str, required=True)
+        self.add_argument('-r', '--regenerate', action='store_true', help=const.HELP_REGENERATE)
+
+    def _run(self, args: argparse.Namespace) -> int:
+        logger.debug('Entering trestle transform.')
+        log.set_log_level_from_args(args)
+        trestle_root = args.trestle_root  # trestle root is set via command line in args. Default is cwd.
+
+        try:
+            return self.transform(
+                trestle_root, args.type, args.transform, args.input, args.transform_by, args.output, args.regenerate
+            )
+        except Exception as e:
+            logger.error(f'The transform operation failed with error: {e}')
+            logger.debug(traceback.format_exc())
+            return 1
+
+    def transform(
+        self,
+        trestle_root: pathlib.Path,
+        model_type: str,
+        transform: str,
+        input_name: str,
+        transform_by: str,
+        output_name: str,
+        regenerate: bool
+    ) -> int:
+        """Transform the input file based on the profile and infer the output type."""
+        if model_type == const.MODEL_TYPE_SSP:
+            if transform == const.FILTER_BY_PROFILE:
```

Comment:
Could be checked in one if statement

Reply:
This was intentional because I anticipate more options added here.  I have added a comment.

Reply:
Thought it could be the reason